### ISSUE TITLE
micro-ros-agent multiarch build

### ISSUE THREAD

**Issue by amfern** [micro-ROS/docker#58]
Hello,

I want to run the docker agent on arm devices, docker has an easy way to create multiarch containers, so i was able to cross compile it to arm with the following set of commands.

```
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
docker buildx create --name multiarch --driver docker-container --use
docker buildx inspect --bootstrap
docker buildx build --platform linux/amd64,linux/arm64 -t microros/base:foxy base
docker buildx build --platform linux/amd64,linux/arm64 --push -t amfernus/micro-ros-agent:foxy micro-ROS-Agent
```
can you add it to your CI?

**Comment by pablogs9** [comment]
Please open a PR contributing your changes.

**Comment by amfern** [comment]
I am not sure where to contribute, I don't see any CI code in this repo

**Comment by pablogs9** [comment]
Can this be done in a Dockerfile? maybe you can contribute the Dockerfile and we will automate the build in Dockerhub

**Comment by amfern** [comment]
No it can't be done with docker file its outside of docker file scope, basically it uses the docker multi arch feature that allows to build multiple architecture for the same dockerfile.

From quick check it seems like docker hub doesn't officially support multi arch builds, but it may be possible to make it work.
https://github.com/docker/hub-feedback/issues/1874, But its very much possible with github actions https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md.
IIUC you are currently using dockerhub to build the docker images, would you consider moving to github actions instead?

**Comment by pablogs9** [comment]
We should reconsider that and plan it in our roadmap.

CC: @FranFin @ralph-lange

**Comment by amfern** [comment]
Just a refernce, this is how i can build multiplatform docker on my computer leveraging the buildx qemu build
https://github.com/amfern/docker-1/commit/3d5ba973ecb6ee8cb49895d0682f03bbdc2865b1

**Comment by pablogs9** [comment]
Solved here dd278e88dafe63a8e7f39437063e58ccf0649cb9
